### DISCUSSION TITLE
Extend multiple extension

### DISCUSSION THREAD

**Discussion by piranhoid** [ueberdosis/tiptap#2095] [General Questions] [answered]
Hi,

I would like to add, an add button, on the left of each block 

I have started to do this : 

`const CustomParagraph = Paragraph.extend({
  addNodeView() {
    return ReactNodeViewRenderer(AddButtonToggler)
  },
})`

But like this, it only works on paragraph node, do you know a way to add this on each types of block ? 

Thanks :)

**Comment by cadars** [comment]
I think you're looking for global attributes: https://tiptap.dev/guide/custom-extensions#global-attributes

**Reply by hanspagel** [reply]
Nah, this doesn’t work for Node Views.

@piranhoid Nope, there’s no such feature. I think you can simplify your code though:

```js
const addButtonNodeView = { addNodeView() { return ReactNodeViewRenderer(AddButtonToggler) } }
Paragraph.extend(addButtonNodeView)
```

**Reply by piranhoid** [reply]
and for each types i add an another line ? 

well ok thanks :)

**Reply by piranhoid** [reply]
your code it's not working because i have to add the extended paragraph to the array of extensions, or i did it wrong ?

**Reply by hanspagel** [reply]
Yeah, that wasn’t a full working example. It would probably look more like this:

```js
const addButtonNodeView = { 
  addNodeView() { return ReactNodeViewRenderer(AddButtonToggler) } 
}

const CustomParagraph = Paragraph.extend(addButtonNodeView)

new Editor({
  extensions: [
    CustomParagraph,
    // …
  ],
})
```

**Comment by piranhoid** [comment, marked as answer]
```
  const editor = useEditor({
    extensions: [
      StarterKit,
      Paragraph.extend(addButtonNodeView),
      Heading.extend(addButtonNodeView),
     // and so on for each type of blocks
    ],
```

Well it doesn't look incredible to me but it will do the trick

Thanks guys !

**Reply by kaspnilsson** [reply]
@piranhoid I'm doing something similar -- how did you add the node view without then overriding the way the Paragraph, Heading etc render natively?

**Reply by katerlouis** [reply]
also interested

**Reply by piranhoid** [reply]
I have changed my way  of doing it 

I've added my AddButton inside the tiptap floating menu, and I've set the prop shouldShow of this component with this variable

```
const displayAddButton = ({ editor, state }) => {
  const { selection } = state
  const { $anchor, empty } = selection
  const isRootDepth = $anchor.depth === 1
  const isEmptyTextBlock =
    $anchor.parent.isTextblock && !$anchor.parent.type.spec.code && !$anchor.parent.textContent
  if (
    !empty ||
    !isRootDepth ||
    !isEmptyTextBlock ||
    $anchor.parent.type.name !== 'paragraph' ||
    !editor.isEditable
  ) {
    return false
  }

  return true
}

export default displayAddButton
```

**Reply by strlrd-29** [reply]
Is there a way to extend the component without overriding the way it renders but just using the thing as a wrapper around it?
